pkg/server/apis/v1alpha1/handler/stage.go

```diff
@@ -0,0 +1,108 @@
+package handler
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/caicloud/nirvana/log"
+	"github.com/gorilla/websocket"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/caicloud/cyclone/pkg/apis/cyclone/v1alpha1"
+	"github.com/caicloud/cyclone/pkg/util/cerr"
+	contextutil "github.com/caicloud/cyclone/pkg/util/context"
+	fileutil "github.com/caicloud/cyclone/pkg/util/file"
+	websocketutil "github.com/caicloud/cyclone/pkg/util/websocket"
+)
+
+// POST /apis/v1alpha1/stages/
+// X-Tenant: any
+func CreateStage(ctx context.Context) (*v1alpha1.Stage, error) {
+	s := &v1alpha1.Stage{}
+	err := contextutil.GetJsonPayload(ctx, s)
+	if err != nil {
+		return nil, err
+	}
+
+	return k8sClient.CycloneV1alpha1().Stages(s.Namespace).Create(s)
+}
+
+// POST /apis/v1alpha1/stages/{stage-name}
+// X-Tenant: any
+func GetStage(ctx context.Context, name, namespace string) (*v1alpha1.Stage, error) {
+	if namespace == "" {
+		namespace = "default"
+	}
+	return k8sClient.CycloneV1alpha1().Stages(namespace).Get(name, metav1.GetOptions{})
+}
+
+// GET /workflowruns/{workflowrun-name}/stages/{stage-name}/streamlogs?container-name=c0
+// ReceivePipelineRecordLogStream receives real-time log of workflowrun stage.
+func ReceivePipelineRecordLogStream(ctx context.Context, workflowrun, stage, container string) error {
```

Comment:
Please use `workflow` and `stage` instead of `pipeline`. Also fix in other places.